include/hc_printf.hpp

```diff
@@ -64,40 +64,31 @@ enum PrintfError {
 
 static inline PrintfPacket* createPrintfBuffer(hc::accelerator& a, const unsigned int numElements) {
   PrintfPacket* printfBuffer = NULL;
-  if (numElements > 6) {
+  if (numElements > 5) {
     printfBuffer = hc::am_alloc(sizeof(PrintfPacket) * numElements, a, amHostCoherent);
 
     // initialize the printf buffer header
     printfBuffer[0].type = PRINTF_BUFFER_SIZE;
     printfBuffer[0].data.ui = numElements;
-    printfBuffer[1].type = PRINTF_BUFFER_CURSOR;
-    printfBuffer[1].data.ui = 5;
-    printfBuffer[2].type = PRINTF_STRING_BUFFER;
-    printfBuffer[2].data.ptr = hc::am_alloc(sizeof(char) * numElements * 12, a, amHostCoherent);
-    printfBuffer[3].type = PRINTF_STRING_BUFFER_SIZE;
-    printfBuffer[3].data.ui = numElements * 12;
-    printfBuffer[4].type = PRINTF_STRING_BUFFER_CURSOR;
-    printfBuffer[4].data.ui = 0;
+    printfBuffer[1].type = PRINTF_STRING_BUFFER;
+    printfBuffer[1].data.ptr = hc::am_alloc(sizeof(char) * numElements * 12, a, amHostCoherent);
+    printfBuffer[2].type = PRINTF_STRING_BUFFER_SIZE;
+    printfBuffer[2].data.ui = numElements * 12;
+
+    // Combine offsets into 1 atomic element to maintain order and atomicity
+    printfBuffer[3].type = PRINTF_OFFSETS;
+    printfBuffer[3].data.ai[0] = 4;
+    printfBuffer[3].data.ai[1] = 0;
```

Comment:
Could we create names for different header indices?

Reply:
Sorry, what did you mean create names? Right now, different header indices have a type member that aligns with enum PrintfPacketDataType.

Reply:
right... could we re-arrange the PrintPacketDataType enum to make the code more readable:

```
enum PrintfPacketDataType {
  PRINTF_BUFFER_SIZE = 0,
  PRINTF_STRING_BUFFER,
...
};

...

printfBuffer[PRINTF_BUFFER_SIZE].type = PRINTF_BUFFER_SIZE;
printfBuffer[PRINTF_BUFFER_SIZE].data.ui = numElements;
```